x-pack/plugins/security_solution/public/timelines/components/edit_data_provider/index.tsx

```diff
@@ -53,9 +53,10 @@ interface Props {
   isExcluded: boolean;
   onDataProviderEdited: OnDataProviderEdited;
   operator: QueryOperator;
-  providerId: string;
+  providerId?: string;
```

Comment:
I am worried about that since I think we are using `providerId` for building draggableId

Reply:
I know, but as we allow now to create data providers manually we don't know what the `providerId` will be without knowing which field and what value is has, that's why I've added
https://github.com/elastic/kibana/pull/67837/files/12e2020a5725aee435e5be7c499519395dfab2c5#diff-6bd770c8a68d1d4c5743a5e6178337f9R76

and used that here
https://github.com/elastic/kibana/pull/67837/files/12e2020a5725aee435e5be7c499519395dfab2c5#diff-6bd770c8a68d1d4c5743a5e6178337f9R162

So once the user chose the field and value we update the providerId